mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -28,144 +31,517 @@ impl IndexedDbNftStorage {
 
-    async fn is_initialized(&self, _chain: &Chain) -> MmResult<bool, Self::Error> { todo!() }
+    async fn is_initialized(&self, _chain: &Chain) -> MmResult<bool, Self::Error> { Ok(true) }
 
     async fn get_nft_list(
         &self,
-        _chains: Vec<Chain>,
-        _max: bool,
-        _limit: usize,
-        _page_number: Option<NonZeroUsize>,
+        chains: Vec<Chain>,
+        max: bool,
+        limit: usize,
+        page_number: Option<NonZeroUsize>,
     ) -> MmResult<NftList, Self::Error> {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftListTable>().await?;
+        let mut nfts = Vec::new();
+        for chain in chains {
+            let items = table.get_items("chain", chain.to_string()).await?;
+            for (_item_id, item) in items.into_iter() {
+                let nft_detail = nft_details_from_item(item)?;
+                nfts.push(nft_detail);
+            }
+        }
+        Self::take_nft_according_to_paging_opts(nfts, max, limit, page_number)
     }
 
-    async fn add_nfts_to_list<I>(&self, _chain: &Chain, _nfts: I, _last_scanned_block: u32) -> MmResult<(), Self::Error>
+    async fn add_nfts_to_list<I>(&self, chain: &Chain, nfts: I, last_scanned_block: u64) -> MmResult<(), Self::Error>
     where
         I: IntoIterator<Item = Nft> + Send + 'static,
         I::IntoIter: Send,
     {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let nft_table = db_transaction.table::<NftListTable>().await?;
+        let last_scanned_block_table = db_transaction.table::<LastScannedBlockTable>().await?;
+        for nft in nfts {
+            let nft_item = NftListTable::from_nft(&nft)?;
+            nft_table.add_item(&nft_item).await?;
+        }
+        let last_scanned_block = LastScannedBlockTable {
+            chain: chain.to_string(),
+            last_scanned_block: BeBigUint::from(last_scanned_block),
+        };
+        last_scanned_block_table
+            .replace_item_by_unique_index("chain", chain.to_string(), &last_scanned_block)
+            .await?;
+        Ok(())
     }
 
     async fn get_nft(
         &self,
-        _chain: &Chain,
-        _token_address: String,
-        _token_id: BigDecimal,
+        chain: &Chain,
+        token_address: String,
+        token_id: BigDecimal,
     ) -> MmResult<Option<Nft>, Self::Error> {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftListTable>().await?;
+        let index_keys = MultiIndex::new(NftListTable::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&token_address)?
+            .with_value(token_id.to_string())?;
```

Comment:
I think there should be aline break after this line

Reply:
Done